MediaBrowser.MediaEncoding/Encoder/MediaEncoder.cs

```diff
@@ -1137,13 +1137,11 @@ public IReadOnlyList<string> GetPrimaryPlaylistM2tsFiles(string path)
             var validPlaybackFiles = _blurayExaminer.GetDiscInfo(path).Files;
 
             // Get all files from the BDMV/STREAMING directory
-            var directoryFiles = _fileSystem.GetFiles(Path.Join(path, "BDMV", "STREAM"));
-
             // Only return playable local .m2ts files
-            return directoryFiles
-                .Where(f => validPlaybackFiles.Contains(f.Name, StringComparer.OrdinalIgnoreCase))
+            return validPlaybackFiles
+                .Select(f => _fileSystem.GetFileInfo(Path.Join(path, "BDMV", "STREAM", f)))
```

Comment:
Won't this read every file individually?

Reply:
One fstat per file, yes
Compared to one opendir and a readdir for each file in that directory

Reply:
Is that how EnumerateFiles works?

Reply:
https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/IO/Enumeration/FileSystemEnumerator.Unix.cs,180